src/components/sbb-form-field/sbb-form-field.tsx

```diff
@@ -0,0 +1,180 @@
+   */
+  private _idError: string;
+
+  /**
+   * @internal
+   * It is used internally to get the `input` slot.
+   */
+  private _input: HTMLInputElement;
+
+  /**
+   * @internal
+   * Listen the changes on `readonly` and `disabled` attributes of `<input>`.
+   */
+  private _formFieldAttributeObserver = new MutationObserver(this._onAttributesChange.bind(this));
+
+  public componentWillLoad(): void {
+    this._idError = this._element?.querySelector('[slot="error"]')?.getAttribute('id');
+    if (this._idError) {
+      this._element.querySelector('[slot="input"]').setAttribute('aria-describedby', this._idError);
+    }
+
+    const input = this._element.querySelector('[slot="input"]');
+    this._readonly = input.hasAttribute('readonly');
+    this._disabled = input.hasAttribute('disabled');
+
+    this._formFieldAttributeObserver.observe(input, {
+      attributes: true,
+      attributeFilter: ['readonly', 'disabled'],
+    });
+  }
+
+  public disconnectedCallback(): void {
+    this._formFieldAttributeObserver.disconnect();
+  }
+
+  /**
+   * @private
+   * It is used internally to assign the attributes of `<input>` to `_id` and `_input`.
+   */
+  private _onSlotInputChange(): void {
+    if (!this._element?.querySelector('[slot="input"]')?.getAttribute('id')) {
+      this._element
+        .querySelector('[slot="input"]')
+        .setAttribute('id', `sbb-form-field-input-${nextId++}`);
+    }
+
+    const inputCssClass = this._element?.querySelector('[slot="input"]')?.getAttribute('class');
+
+    this._element
+      .querySelector('[slot="input"]')
+      .setAttribute('class', `${inputCssClass} form-field--size-${this.size}`);
+
+    this._id = this._element.querySelector('[slot="input"]').getAttribute('id');
+
+    this._input = this._element.querySelector('[slot="input"]') as HTMLInputElement;
+  }
+
+  /**
+   * @private
+   * It is used internally to set the focus in the `<input>`.
+   */
+  private _setFocus(): void {
+    this._input.focus();
+  }
+
+  /**
+   * @param mutationsList The list of the attributes
+   * @private
+   * It is used internally to bind on the `MutationObserver`.
+   */
+  private _onAttributesChange(mutationsList: MutationRecord[]): void {
+    for (const mutationRecord of mutationsList) {
+      if (mutationRecord.type !== 'attributes') {
+        return;
+      }
+      const inputEl = mutationRecord.target as HTMLInputElement;
+      this._disabled = inputEl.hasAttribute('disabled');
+      this._readonly = inputEl.hasAttribute('readonly');
+    }
+  }
+
+  public render(): JSX.Element {
+    const optional = this.optional ? '(optional)' : '';
+    const cssClassErrorSpace = `form-field--error-space-${this.errorSpace}`;
+    const cssSizeClass = `form-field--size-${this.size}`;
+    const cssClassSlotPrefix = this._element.querySelector('[slot="prefix"]') ? 'form--prefix' : '';
```

Comment:
I don't think this would update, if a prefix or suffix is added or removed?
Could this maybe also be defined by `slotchange`? (e.g. `onslotchange={(event) => this._hasPrefix = (event.target as HTMLSlot).assignedElements().length > 0}`)